src/System.Net.Http/src/System/Net/Http/Managed/AuthenticationHandler.cs

```diff
@@ -53,17 +49,26 @@ private bool TrySetBasicAuthToken(HttpRequestMessage request)
 
                 foreach (AuthenticationHeaderValue h in authenticateValues)
                 {
-                    // We only support Basic auth, ignore others
-                    if (h.Scheme == Basic)
+                    // We only support Basic and digest auth, ignore others
```

Comment:
If the server returns with a Unauthorized response when you have tried to authenticate with a prior response, check to see if the new response has stale set. If it does, allow to try authentication again. If the server returns stale again, something odd is happening so you should stop there and return the unauthorized response to the calling code.